packages/components/dashboard/resource-providers-card/resource-providers-card.tsx

```diff
@@ -11,105 +11,63 @@ import {
-  Number(_.get(provider, 'value[1]', null));
-
-const filterProviders = (allProviders: ProviderType): string[] => {
-  return _.keys(allProviders).filter((provider) => allProviders[provider] > 0);
-};
-
-const createProvidersList = (data: PrometheusResponse): ProviderType => {
-  const providers = _.get(data, 'data.result', null);
-  const providersList: ProviderType = {};
-  if (_.isNil(providers)) return {};
-  providers.forEach((provider) => {
-    providersList[getProviderType(provider)] = getProviderCount(provider);
-  });
-  return providersList;
-};
+const ResourceProvidersItem: React.FC<ResourceProvidersRowProps> =
+  // eslint-disable-next-line react/display-name
+  React.memo(({ title, count }) => (
+    <div className="co-inventory-card__item">
+      <div
+        className="nb-resource-providers-card__row-title"
+        data-test="nb-resource-providers-card"
+      >{`${count} ${title}`}</div>
+    </div>
+  ));
 
 const ResourceProviders: React.FC<{}> = () => {
   const { t } = useTranslation();
 
-  const [providersTypesResult, providersTypesError, providersTypesLoading] =
-    useCustomPrometheusPoll({
-      query: RESOURCE_PROVIDERS_QUERY.PROVIDERS_TYPES,
-      endpoint: 'api/v1/query' as any,
-      basePath: MCG_MS_PROMETHEUS_URL,
-    });
-  const [
-    unhealthyProvidersTypesResult,
-    unhealthyProvidersTypesError,
-    unhealthyProvidersTypesLoading,
-  ] = useCustomPrometheusPoll({
-    query: RESOURCE_PROVIDERS_QUERY.UNHEALTHY_PROVIDERS_TYPES,
-    endpoint: 'api/v1/query' as any,
-    basePath: MCG_MS_PROMETHEUS_URL,
-  });
-  const [resourcesLinksResponse, resourcesLinksError, resourcesLinksLoading] =
-    useCustomPrometheusPoll({
-      query: RESOURCE_PROVIDERS_QUERY.RESOURCES_LINK_QUERY,
-      endpoint: 'api/v1/query' as any,
-      basePath: MCG_MS_PROMETHEUS_URL,
-    });
-  const error =
-    !!providersTypesError ||
-    !!unhealthyProvidersTypesError ||
-    !!resourcesLinksError;
-
-  const noobaaResourcesLink = getMetric(resourcesLinksResponse, 'resources');
-
-  const allProviders = createProvidersList(providersTypesResult);
-  const unhealthyProviders = createProvidersList(unhealthyProvidersTypesResult);
+  const [dataResources, dataResourcesLoaded, dataResourcesError] =
+    useK8sWatchResource<NamespaceStoreKind[]>(nameSpaceStoreResource);
 
-  const providerTypes = filterProviders(allProviders);
-  const hasProviders = !_.isEmpty(allProviders);
-  const isLoading =
-    providersTypesLoading ||
-    unhealthyProvidersTypesLoading ||
-    resourcesLinksLoading;
+  const allProvidersMap = React.useMemo(
+    () =>
+      dataResourcesLoaded && !dataResourcesError
+        ? dataResources?.reduce(
+            (drMap: AllProvidersMap, dataResource: NamespaceStoreKind) => {
+              const providerType =
+                NOOBAA_PROVIDER_MAP[dataResource?.spec?.type];
+              dataResource?.status?.phase === NS_PROGRESS.READY &&
```

Comment:
are we mentioning anywhere that we are showing only the READY state in the resource provider card? like using some tooltip or something.

Reply:
we are not yet... and I think we should be... let's confirm this with @cloudbehl on our next sync up and if needed will add that to the same PR which will be making other text related changes to various components...

Reply:
cool